node/node.go

```diff
@@ -251,6 +275,39 @@ func (n *NodeController) Ready() <-chan struct{} {
 	return n.chReady
 }
 
+// pingLoop does one thing. It pings the provider. It does this until context is cancelled.
+func (n *NodeController) pingLoop(ctx context.Context) {
+	realCheckFunc := func(ctx context.Context) {
+		ctx, span := trace.StartSpan(ctx, "node.pingNode")
+		defer span.End()
+		err := n.p.Ping(ctx)
+		n.lastPingErrorLock.Lock()
+		n.lastPingError = err
```

Comment:
I figure errors best be left uncopied. Hopefully no one is doing anything creative here.